### PR TITLE
fix kubectl e2e test to avoid alpha versions unless needed

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/kubernetes/kubernetes/issues/35429

The clientset was confused by a bad choice of default version that was forced.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35456)

### CODE REVIEW

**Comment by deads2k** [comment]
@ncdc @eparis I got green gkes. ptal.  If everything goes green, let's get this (or if you have an alternate fix) in.

**Comment by deads2k** [comment]
@k8s-oncall This fixes the test blocking http://submit-queue.k8s.io/#/e2e on my local machine.

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -169,10 +169,62 @@ func runKubectlRetryOrDie(args ...string) string {
...
 	f := framework.NewDefaultGroupVersionFramework("kubectl", BatchV2Alpha1GroupVersion)
 
+	var c clientset.Interface
+	var ns string
+	BeforeEach(func() {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+	})
+
+	// Customized Wait  / ForEach wrapper for this test.  These demonstrate the
```
Comment (ncdc) [inline test/e2e/kubectl.go]: fragment?

**Comment by ncdc** [comment]
lgtm

File: test/e2e/kubectl.go (hunk 1 above)
Comment (deads2k) [inline test/e2e/kubectl.go]: > fragment?

dead comment.  Worth losing green on a blocking flake fix for?

**Comment by foxish** [comment]
@ncdc @deads2k All good to go ahead and merge?

File: test/e2e/kubectl.go (hunk 1 above)
Comment (ncdc) [inline test/e2e/kubectl.go]: No, we can deal with it later

**Comment by janetkuo** [comment]
LGTM

**Comment by ncdc** [comment]
Go for it

**Pull request merged**